core/shared/platform/common/posix/posix_socket.c

```diff
@@ -176,3 +177,100 @@ os_socket_inet_network(const char *cp, uint32 *out)
     *out = ntohl(inet_addr(cp));
     return BHT_OK;
 }
+
+static int
+getaddrinfo_error_to_errno(int error)
+{
+    switch (error) {
+        case EAI_AGAIN:
+            return EAGAIN;
+        case EAI_FAIL:
+            return EFAULT;
+        case EAI_MEMORY:
+            return ENOMEM;
+        case EAI_SYSTEM:
+            return errno;
+        default:
+            return EINVAL;
+    }
+}
+
+static int
+is_addrinfo_supported(struct addrinfo *info)
+{
+    return
+        // Allow only IPv4 and IPv6
+        (info->ai_family == AF_INET || info->ai_family == AF_INET6)
+        // Allow only UDP and TCP
+        && (info->ai_socktype == SOCK_DGRAM || info->ai_socktype == SOCK_STREAM)
+        && (info->ai_protocol == IPPROTO_TCP
+            || info->ai_protocol == IPPROTO_UDP);
+}
+
+int
+os_addr_resolve(const char *host, const char *service, uint8_t *hint_is_tcp,
+                uint8_t *hint_is_ipv4, wamr_addr_info_t *addr_info,
+                size_t addr_info_size, size_t *max_info_size)
+{
+    struct addrinfo hints, *res, *result;
+    int hints_enabled = hint_is_tcp || hint_is_ipv4;
+    int ret;
+    size_t pos = 0;
+
+    if (hints_enabled) {
+        memset(&hints, 0, sizeof(hints));
```

Comment:
Had better memset hints before L220 to avoid unexpected behavior.

Reply:
what unexpected behavior do you anticipate here? `hints` variable is only being used when `hints_enabled` is set to true:
```
getaddrinfo(host, service, hints_enabled ? &hints : NULL, &result);
```

Reply:
Yes, not sure whether it will impacts `getaddrinfo(host, service, hints_enabled ? &hints : NULL, &result);` in L230. It should be OK to memset here, we are just used to initialize the structure at the beginning, e.g. `struct addrinfo hints = { 0 } ` at the definition.